### DISCUSSION TITLE
Disable Zed AI in the editor

### DISCUSSION THREAD

**Discussion by appelgriebsch** [zed-industries/zed#16750] [Feature Requests]
Hey,

with the recent introduction of the Zed AI in the editor I see some issues with running it in an organizational setup as there might be company restrictions in place for using (public) AI tools in the daily workflow.

As such my question here: is there a way to disable the Zed AI functionality from the editor completely?

**Comment by alphabitserial** [comment]
Agreed. Further, I expect that as new "AI features" are added, even if they can be individually disabled at runtime, they're likely to ship enabled by default. Therefore, I think a compile-time flag for disabling all of the "AI features" would be ideal.

**Reply by twokilohertz** [reply]
Fingers crossed for this improvement 🤞

**Comment by rohitshriwas** [comment]
In settings, you can disable both things with:
```JSONC
{
  "features": {
    // Which inline completion provider to use.
    "inline_completion_provider": "none"
  },
  "assistant": {
    "enabled": false,
    "version": "2"
  },
}
```

**Reply by polarfish** [reply]
In March 2025 we should use  `"edit_prediction_provider": "none"` instead of `"inline_completion_provider": "none"`

**Reply by applejag** [reply]
And `"assistant"` has been changed to `"agent"`

```json
{
  "features": {
    "edit_prediction_provider": "none"
  },
  "agent": {
    "enabled": false,
    "version": "2"
  },
}
```

**Reply by nkapila6** [reply]
Thanks apple, you're a life saver.

**Reply by sgriff96** [reply]
Do we know if this will completely disable AI from collecting any data from our codebase? Or is it just turning off the ai suggestions but kind of leaving the AI on in the background?

**Reply by applejag** [reply]
> Do we know if this will completely disable AI from collecting any data from our codebase? Or is it just turning off the ai suggestions but kind of leaving the AI on in the background?

Not unless we dig through the codebase, and I don't want to read a bunch of Rust

**Reply by iLillie** [reply]
Would be great to have a single config option to disable all AI options, including outgoing calls for models.

I did some checking with Charles Proxy and found the following language_models are being called by default when you launch the application.
* open_router (fetching models) 
  * GET https://openrouter.ai/api/v1/models
* ollama (fetching tags)
  * GET http://localhost:11434/api/tags
* lmstudio (fetching models)
  * GET http://localhost:1234/api/v0/models

If you would like to make sure no calls are going out to any AI endpoints update config with

```json
  "features": {
    "edit_prediction_provider": "none"
  },
  "agent": {
    "enabled": false,
    "version": "2"
  },
  "language_models": {
    "anthropic": {
      "version": "1",
      "api_url": ""
    },
    "google": {
      "api_url": ""
    },
    "ollama": {
      "api_url": ""
    },
    "openai": {
      "version": "1",
      "api_url": ""
    },
    "open_router": {
      "api_url": ""
    },
    "lmstudio": {
      "api_url": ""
    },
    "deepseek": {
      "api_url": ""
    },
    "mistral": {
      "api_url": ""
    }
  }
```

**Comment by spuxx-dev** [comment]
Could we possibly get a keymap shortcut for this? `editor::ToggleEditPrediction` doesn't seem to do the trick, it simply hides the currently displayed prediction, but does not disable any further ones.

**Comment by ahx** [comment]
A global setting to deactivate all AI / LLM interactions would be great.

**Reply by Cyncrovee** [reply]
> A global setting to deactivate all AI / LLM interactions would be great.

This is definitely a must-have, at least in my opinion. Just one setting to disable *all* AI stuff in the editor- it would be a lot easier than having to hunt down every AI option and manually disable them.

**Reply by niltonheck** [reply]
Anyone working actively on this? It's a must-have to me too. I might have some time to take a look into it.

**Comment by keithluchtel** [comment]
Is there the capability to disable AI features on per-project basis? Trying to edit the settings in the project settings file doesn't appear to be working for me.

**Reply by nyabinary** [reply]
Per file too, or a snooze function would be nice

**Comment by recursiveAdaption** [comment]
Does anyone know if there's any AI telemetry or background running services even with the suggested changes in settings?

I just discovered Zed and I want to use it but only if I assurances that I'm not being constantly monitored.

**Reply by mgsloan** [reply]
@recursiveAdaption When you start Zed for the first time it will ask about telemetry preferences

<img width="2205" height="1377" alt="image" src="https://github.com/user-attachments/assets/e21e0d4c-0ca2-4154-8148-9f9999e8093f" />

Or add this to settings.json:

```
"telemetry": {
    "diagnostics": false,
    "metrics": false
},
```

**Comment by sudo-nano** [comment]
I would also love a single setting toggle to disable all AI/LLM features. I use Zed because it's one of the only non-web app IDEs out there, not because I want the AI features.

**Comment by Calinou** [comment]
This was added to Preview today and will be in Stable soon: https://zed.dev/blog/disable-ai-features

**Reply by keithluchtel** [reply]
I see in the blog it mentions it as a global setting, but didn't see if it mentions being able to override or configure per project in a project config file.

**Reply by injust** [reply]
This partially resolves https://github.com/zed-industries/zed/issues/31346 -- it removes the `agent` and `assistant` commands, but not `copilot` commands.

For it to be a complete solution, it would ideally be more granular (personally I would like to disable most AI functionality, but keep inline completions).

**Comment by JosephTLyons** [comment]
Hey all, as mentioned by @injust, we recently landed a [PR](https://github.com/zed-industries/zed/pull/34896) that added the following setting:

`"disable_ai": true`

This disables all things AI related in Zed, at run time. This is in [Preview v0.197](https://zed.dev/releases/preview?g=72) as of today, and will be in v0.197 Stable next Wednesday (07/30/2025).

https://zed.dev/blog/disable-ai-features

**Reply by jdiago** [reply]
This option seems to not completely work. While it does hide all the ai features from the UI, edit prediction is still working. This screenshot is from:
```
Zed 0.197.3 
002b8b9f5a5df0c991bfe7fcbd755737f1d55d82
```
<img width="1167" height="670" alt="image" src="https://github.com/user-attachments/assets/b8d6bbe4-acd0-4b63-8c72-7892f77977c0" />

**Reply by mgsloan** [reply]
@jdiago  Ah dang, thanks for reporting!  Should be fixed by #35513

**Comment by Tirmisee** [comment]
How can I personalize Zed so that I use only GitHub Copilot and Codex, without showing any other AI options in the UI? I want to remove all other AI integrations and keep only GitHub Copilot and Codex, just like I did in VS Code.

**Comment by keithluchtel** [comment]
Would be great if there was an option to disable AI on a project-by-project basis. Some projects I want to use AI, and others I don't.

**Reply by mgsloan** [reply]
You can!  Put `{ "disable_ai": true }` in `.zed/settings.json` at the root of the project.
